admin/test/admin_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// This file has the same content as admin_test.go, difference is that this is
+// in another package, and it imports "xds", so we can test that csds is
+// registered when xds is imported.
+
+package test_test
+
+import (
+	"testing"
+
+	"google.golang.org/grpc/admin/test"
+	"google.golang.org/grpc/codes"
+	_ "google.golang.org/grpc/xds"
+)
+
+func TestRegisterWithCSDS(t *testing.T) {
+	if err := test.RunRegisterTests([]test.RunAndCode{
```

Comment:
I think it might be nicer to pass `t` to `RunRegisterTests` so it will print the proper line where the failure happened.

You can also do a `t.Run` then.

Reply:
Done.

How to do `t.Run`?

Reply:
```go
t.Run("channelz", func(t *testing.T) {
	if err := RunChannelz(conn); status.Code(err) != ec.ChannelzCode {
		t.Fatalf("%s test failed with error %v, want code %v", "channelz", err, ec.ChannelzCode)
	}
})
t.Run("csds", func(t *testing.T) {
	if err := RunCSDS(conn); status.Code(err) != ec.CSDSCode {
		t.Fatalf("%s test failed with error %v, want code %v", "CSDS", err, ec.CSDSCode)
	}
})
```

Reply:
Done. Maybe we don't need `"%s test failed"` any more? But I kept them.